src/Http/Routing/src/EndpointNameAddressScheme.cs

```diff
@@ -66,21 +71,20 @@ public IEnumerable<Endpoint> FindEndpoints(string address)
         }
 
         // OK we need to report some duplicates.
-        var duplicates = endpoints
-            .GroupBy(GetEndpointName)
```

Comment:
Was this group by useless or just changes the order the endpoints are written out?

Reply:
They weren't useless. It was just re-doing what the above Dictionary was doing. Previously, when a duplicate was found, it stopped filling the Dictionary, and then re-did the grouping here. I just changed it to continue adding duplicates to the Dictionary, and then use the Dictionary to build the exception message.